packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -420,11 +432,20 @@ export async function* OpenAIChatModel(
       if (delta.function_call.arguments) {
         currentMessage.function_call.arguments += delta.function_call.arguments;
       }
+      if (props.experimental_streamFunctionCallOnly) {
+        yield JSON.stringify({
+          ...currentMessage.function_call,
+          arguments: untruncateJson.default(currentMessage.function_call.arguments ?? '{}'),
+        });
+      }
     }
   }
 
   logger.debug({ message: currentMessage }, 'Finished createChatCompletion');
 
+  if (props.experimental_streamFunctionCallOnly) {
+    return JSON.stringify(currentMessage.function_call);
+  }
```

Comment:
@petersalas now that you see what I had in mind, lmk what you think.

Reply:
it's about what I expected :)